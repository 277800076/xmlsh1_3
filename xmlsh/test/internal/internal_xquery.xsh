# test of xquery

D1=<[
document {
<root a1="value1">
  <foo>
    <bar a2="value2"/>
    text
   </foo>
</root>
}
]>

cd ../../samples/data

#xpath on stdin
echo Xquery on stdin
xquery //AUTHOR < books.xml

#xpath on standard file
echo Xquery on file
xquery -i books.xml //AUTHOR

#xpath with xml expression
echo Xquery on xml expression
xquery -i $D1 //bar

#xpath with substitued variables and no input
xquery -n -v 'declare variable $x external ; $x/foo' x <[<bar><foo>text</foo></bar>]>

# more complicated xquery

xquery -f ../query/books.xquery -i books.xml




