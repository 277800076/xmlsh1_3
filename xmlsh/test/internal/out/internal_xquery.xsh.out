Xquery on stdin
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jane Austen</AUTHOR><AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Charlotte Brontë</AUTHOR><AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Thomas Hardy</AUTHOR><AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Thomas Hardy</AUTHOR><AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jasper Fforde</AUTHOR><AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jasper Fforde</AUTHOR>Xquery on file
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jane Austen</AUTHOR><AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Charlotte Brontë</AUTHOR><AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Thomas Hardy</AUTHOR><AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Thomas Hardy</AUTHOR><AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jasper Fforde</AUTHOR><AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jasper Fforde</AUTHOR>Xquery on xml expression
<bar a2="value2"/><foo>text</foo><book><author><AUTHOR>Jasper Fforde</AUTHOR></author><title><TITLE>The Eyre Affair</TITLE></title></book><book><author><AUTHOR>Thomas Hardy</AUTHOR></author><title><TITLE>Jude the Obscure</TITLE></title></book><book><author><AUTHOR>Charlotte Brontë</AUTHOR></author><title><TITLE>Wuthering Heights</TITLE></title></book><book><author><AUTHOR>Jasper Fforde</AUTHOR></author><title><TITLE>The Big Over Easy</TITLE></title></book><book><author><AUTHOR>Jane Austen</AUTHOR></author><title><TITLE>Pride and Prejudice</TITLE></title></book><book><author><AUTHOR>Thomas Hardy</AUTHOR></author><title><TITLE>Tess of the d'Urbervilles</TITLE></title></book>